fenix/app/src/main/java/org/mozilla/fenix/components/appstate/shopping/ShoppingState.kt

```diff
@@ -7,11 +7,28 @@ package org.mozilla.fenix.components.appstate.shopping
 /**
  * State for shopping feature that's required to live the lifetime of a session.
  *
- * @property productsInAnalysis Set of product ids that are currently being analysed or were being
+ * @property productsInAnalysis Set of product urls that are currently being analysed or were being
  * analysed when the sheet was closed.
  * @property shoppingSheetExpanded Boolean indicating if the shopping sheet is expanded and visible.
+ * @property productCardState Map of product url to [CardState] that contains the state of different
+ * cards in the shopping sheet.
  */
 data class ShoppingState(
     val productsInAnalysis: Set<String> = emptySet(),
     val shoppingSheetExpanded: Boolean? = null,
-)
+    val productCardState: Map<String, CardState> = emptyMap(),
```

Comment:
Do we foresee any potential fragility with using the URL as the key here? In other words, is it guaranteed that the URL we use from the Fakespot service is the same between API calls?

Reply:
Good question. For the case you mentioned, this works fine. The url we use is the `selectedTabUrl`.

I am concerned about the cases when the url changes because of some query params but the product id is the same, aka same pdp. This is the same for persisting reanalysis states, which also use product urls in a set. So far, we haven't encountered any issues with it. But if we do, we could use an object with vendor domain and product id instead. While working on persisting renalaysis, that was complicating it before we even encountered an issue with the url approach. Maybe I'm overthinking this, however, since we don't know all the different product pages, it's a bit hard to optimize this solution based on the current information. wdyt?

Reply:
I also had a concern about pages changing the URL whenever the user clicks on site elements.

Yeah, this may be part of a larger conversation as this feature continues to iterate. Incorporating the Product ID could be a potential candidate. Tab ID could be an option as well; it just wouldn't cover the scenario that a user has the product open in more than tab. 

I'm cool with landing as-is, since we have this on our radar and have a couple hypothetical solutions in case issues do arise once this feature is truly in the wild.